pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -67,14 +80,15 @@ def new_process(self, process_idx: int, trainer) -> None:
             trainer.progress_bar_callback.disable()
 
         self.model_to_device()
+        trainer.accelerator_backend.setup_optimizers(trainer)
         self.barrier()
 
         if trainer.testing:
             results = trainer.run_test()
         else:
             results = trainer.train()
 
-        self.__save_end_of_training_weights(self.lightning_module)
+        self.__save_end_of_training_weights(self.lightning_module, trainer)
```

Comment:
dont think `trainer ` arg is used. Maybe we can omit it from the method signature itself.